internal/builder/inspect.go

```diff
@@ -0,0 +1,147 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pubbldr "github.com/buildpacks/pack/builder"
+
+	"github.com/buildpacks/pack/internal/dist"
+
+	"github.com/buildpacks/pack/config"
+)
+
+type Info struct {
+	Description     string
+	StackID         string
+	Mixins          []string
+	RunImage        string
+	RunImageMirrors []string
+	Buildpacks      []dist.BuildpackInfo
+	Order           pubbldr.DetectionOrder
+	BuildpackLayers dist.BuildpackLayers
+	Lifecycle       LifecycleDescriptor
+	CreatedBy       CreatorMetadata
+}
+
+type Inspectable interface {
+	Label(name string) (string, error)
+}
+
+type InspectableFetcher interface {
+	Fetch(ctx context.Context, name string, daemon bool, pullPolicy config.PullPolicy) (Inspectable, error)
+}
+
+type LabelManagerFactory interface {
+	BuilderLabelManager(inspectable Inspectable) LabelInspector
+}
+
+type LabelInspector interface {
```

Comment:
Could you give some additional clarification on this point? Each of the methods `Metadata` `StackID`, etc indicates the  type of info being retrieved.

Reply:
My initial instinct was to propose something like `BuilderInspector` since it's the information of a builder that you are trying to inspect as opposed to where the data is stored (Label). Upon further inspection, no pun intended, I do see that it's part of the builder package so it's less ideal to have `builder.BuilderInspector` vs `builder.LabelInspector` and `builder.Inspector` is already taken. There is probably still a bit of clean up that can be done here but given that this is an internal package and it's simply semantics I'm okay with pushing this forward as is.